qemu/tests/x86_cpu_flag_disable.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+from virttest import env_process
+from virttest import error_context
+
+from provider.cpu_utils import check_cpu_flags
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu flags.
+    1) Check if current flags are in the supported lists, if no, cancel test
+    2) Otherwise, boot guest with the cpu flags disabled
+    3) Check cpu flags inside guest(only for linux guest)
+    4) Check kvmclock inside guest(only for linux guest)
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    flag = params["flags"]
+    check_host_flags = params.get_boolean("check_host_flags")
+    if check_host_flags:
+        check_cpu_flags(params, flag, test)
+
+    params["start_vm"] = "yes"
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+
+    vm = env.get_vm(vm_name)
+    cmd = params["check_flag_cmd"]
+    error_context.context("Try to log into guest", logging.info)
+    session = vm.wait_for_login()
+    vm_flags_out = session.cmd_output(cmd).split()
+    if flag in vm_flags_out:
+        test.fail("Flag %s shouldn't inside guest" % flag)
```

Comment:
You can call check_cpu_flags() directly by setting 'no_flags'.

Reply:
@huangyum  shouldn't we check the flag that shouldn't be in guest?

Reply:
Yes, so you need set 'no_flags'.

Reply:
Thanks and update